tasks/lzards-backup/src/index.ts

```diff
@@ -268,14 +278,31 @@ export const backupGranule = async (params: {
     cloudfrontEndpoint?: string,
   },
 }) => {
+  let granuleCollection : CollectionRecord;
+  let collectionId: string = '';
   const { granule, backupConfig } = params;
+  const messageGranule = granule as MessageGranuleFromStepOutput;
+  const apiGranule = granule as ApiGranule;
   log.info(`${granule.granuleId}: Backup called on granule: ${JSON.stringify(granule)}`);
   try {
-    const granuleCollection = await getGranuleCollection({
-      collectionName: granule.dataType,
-      collectionVersion: granule.version,
-    });
-    const collectionId = constructCollectionId(granule.dataType, granule.version);
+    if (apiGranule.collectionId) {
+      collectionId = apiGranule.collectionId;
+      const { name, version } = deconstructCollectionId(apiGranule.collectionId);
+      granuleCollection = await getGranuleCollection({
+        collectionName: name,
+        collectionVersion: version,
+      });
+    } else if (messageGranule.dataType && messageGranule.version) {
+      granuleCollection = await getGranuleCollection({
+        collectionName: messageGranule.dataType,
+        collectionVersion: messageGranule.version,
+      });
+      collectionId = constructCollectionId(messageGranule.dataType, messageGranule.version);
+    } else if (!apiGranule.collectionId || !(messageGranule.dataType && messageGranule.version)) {
```

Comment:
What are your thoughts on this: https://github.com/nasa/cumulus/pull/3043/commits/ad477d13003920919959a9deef5c8793b2e4cd09#diff-31ab5953f1d25d2a7e338fb25acc5c7ae53aacc8cde330d0084eb4e00236cf3fR300-R305
```
    } else {
      if (!apiGranule.collectionId) {
        log.error(`${JSON.stringify(granule)}: Granule did not have collectionId and was unable to identify a collection.`);
      } else if (!messageGranule.dataType && !messageGranule.version) {
        log.error(`${JSON.stringify(granule)}: Granule did not have dataType and version and was unable to identify a collection.`);
      }
      throw new CollectionIdentifiersNotProvidedError('[dataType and version] or [collectionId] required.');
    }
```

Reply:
What if `backupGranule` is intended to be called with `dataType` and `version` but (for example) `version` is missing? I think in that case we'd log `Granule did not have collectionId and was unable to identify a collection.` because it also doesn't have a `collectionId`. 

I don't know if that's an actual use case or potential problem. Maybe that's not possible depending on how it's called.

In any case, I'll bet we could simplify the `else` logic here and just log a message similar to the error we're throwing. Instead of:
```
} else {
      if (!apiGranule.collectionId) {
        log.error(`${JSON.stringify(granule)}: Granule did not have collectionId and was unable to identify a collection.`);
      } else if (!messageGranule.dataType && !messageGranule.version) {
        log.error(`${JSON.stringify(granule)}: Granule did not have dataType and version and was unable to identify a collection.`);
      }
      throw new CollectionIdentifiersNotProvidedError('[dataType and version] or [collectionId] required.');
    }
```
what if we did something like:

```
} else {
      log.error(`${JSON.stringify(granule)}: Granule did not have [collectionId] or [dataType and version] and was unable to identify a collection.`);
      throw new CollectionIdentifiersNotProvidedError('[dataType and version] or [collectionId] required.');
    }
```

Reply:
That sounds reasonable to me! 
https://github.com/nasa/cumulus/pull/3043/commits/e55240184f9a102ea07e60e8fdd5559546d1705c#diff-31ab5953f1d25d2a7e338fb25acc5c7ae53aacc8cde330d0084eb4e00236cf3fR301